### DISCUSSION TITLE
[6.14.2] useWatch on a field value always gets its defaultValue after removing other field

### DISCUSSION THREAD

**Discussion by github0013** [react-hook-form/react-hook-form#4081] [General]
https://codesandbox.io/s/festive-pike-j1eg5?file=/src/App.tsx

This form has a regular form value (form name) + field array.

1. append more than 2 fields
2. fill in the field names
3. delete one of the field
4. check the console
5. you will see something like `name: name default (index: 1)` which is coming from useWatch's defaultValue

![festive-pike-j1eg5_-_CodeSandbox](https://user-images.githubusercontent.com/809378/106540535-74b27a00-6543-11eb-89dd-112e937428eb.png)

I am not sure if this is a bug, or I am mis-using these methods.
(handleSubmit's data value gets what I see in the form though...)

- :white_check_mark: `ref={register()}` on fields
- :white_check_mark:  setting `defaultValue` on fields

Did anyone have the same issue?

**Comment by bluebill1049** [comment]
This is a bug, which has been patch. i will release it today.

**Reply by github0013** [reply]
oh, thank you always!

**Reply by github0013** [reply]
Was it this https://github.com/react-hook-form/react-hook-form/issues/2516 that you fixed in 6.15.0?

https://codesandbox.io/s/festive-pike-j1eg5?file=/src/App.tsx
It now gets watch value properly after removing other fields, but useWatch now doesn't return the current value as typed in.

**Reply by bluebill1049** [reply]
oh wow you are right...

**Reply by bluebill1049** [reply]
send an issue i will try to patch it tonight

**Reply by bluebill1049** [reply]
https://codesandbox.io/s/objective-hypatia-kvz2o?file=/src/src/useWatch.ts

**Reply by github0013** [reply]
![objective-hypatia-kvz2o_-_CodeSandbox](https://user-images.githubusercontent.com/809378/106549957-6f5e2b00-6555-11eb-9036-88658404c12c.png)
was it the hotfix?  I still see something wierd.

**Reply by bluebill1049** [reply]
how to reproduce the above?

**Reply by bluebill1049** [reply]
nvm just reproduced it.

**Reply by github0013** [reply]
1. add multiple fields first (don't type in yet)
2. then start typing 
3. probably your fields gets `""` except the last one.

**Reply by bluebill1049** [reply]
have a test this one plz: https://codesandbox.io/s/objective-hypatia-kvz2o

**Reply by github0013** [reply]
![objective-hypatia-kvz2o_-_CodeSandbox](https://user-images.githubusercontent.com/809378/106551430-966a2c00-6558-11eb-908b-c2c6fd2cadaa.png)

It's still the same result on my side.   Only the last one properly reacts if you add multiple fields first, then start typing.

If you add a field, type in, then add another ... then it seems to work as expected.

**Reply by bluebill1049** [reply]
looks like doing a merge with current fields value with field array value solves this problem.
<img width="1098" alt="Screen Shot 2021-02-02 at 3 42 11 pm" src="https://user-images.githubusercontent.com/10513364/106553293-392ca580-656d-11eb-8e26-52986b46ca15.png">

https://codesandbox.io/s/objective-hypatia-kvz2o

honestly, can't wait for 7 to solve those problems in a much easier and elegant way.

**Reply by github0013** [reply]
This fix is going great.  Thanks a bunch!
I can't wait for V7 here too.

**Reply by bluebill1049** [reply]
v7: https://codesandbox.io/s/objective-hypatia-kvz2o

**Comment by bluebill1049** [comment]
also, FYI useFieldArray is getting a lot of improvement in v7: https://github.com/react-hook-form/react-hook-form/discussions/3714

**Reply by github0013** [reply]
Good to know!  I didn't know you were already working on V7.

So if this is a known issue on V6, the workaround I can think of is to use `React.useState` to memo the last watched value and use it as the defaultValue?

https://codesandbox.io/s/admiring-surf-5wigh?file=/src/App.tsx

**Reply by bluebill1049** [reply]
can you try with the latest version? 6.15.0
